scripts/leeds_census_2021.R

```diff
@@ -1,138 +1,132 @@
+    select = c("GEOGRAPHY_CODE", "OBS_VALUE")
+  ) %>%
+    rename(!!sym(.y) := OBS_VALUE, !!geo_col := GEOGRAPHY_CODE))
+  
+  # Load lookup file
+  lookup_data <- read.csv("raw_data/geography_lookup_Dec_2021.csv") %>%
+    rename_with(tolower) %>%  # Convert column names to lowercase
+    select(ends_with("cd"))   # Keep only columns ending with "cd"
+  
+  # Merge all datasets
+  dataframes <- c(list(lookup_data), COB_data, ethnicity_data)
+  merged_data <- reduce(dataframes, left_join, by = geo_col)
+  
+  # Filter for Leeds LAD22CD
+  leeds_data <- merged_data %>%
+    filter(lad22cd == "E08000035") %>%
+    distinct(!!sym(geo_col), .keep_all = TRUE) %>%
+    select(-matches("NMW|ObjectId"))
+  
+  return(leeds_data)
+}
+
+# Call in data for OA level
+leeds_census_data_2021_oa <- get_census_data("TYPE150", "oa21cd")
+
+# Call in data for LSOA level
+leeds_census_data_2021_lsoa <- get_census_data("TYPE151", "lsoa21cd")
+
+# Call in data for MSOA level
+leeds_census_data_2021_msoa <- get_census_data("TYPE152", "msoa21cd")
+
+# Use ONS data on eligible households based on age of child - downloaded on March 7th 2025
+# The ONS groups dependent children in the following age bands:[0,4], [5,9], [10,15], [16,18]
+# In this script eligible children are defined to be in the age bands [0,9]
+
+dep_child_age_data <- drive_get("Lead_Map_Project/UK/predictors/leeds_data/household_dependent_child_age_ons.csv") |>
+  drive_read_string() |>
+  read_csv()
+
+dep_child_age_data <- dep_child_age_data %>%
+  rename(
+    oa21cd = 1,                  # Rename first column
+    dep_child_category = 3          # Rename third column
+  )
+
+lookup_data <- read.csv("raw_data/geography_lookup_Dec_2021.csv") %>%
+  rename_with(tolower) %>%  # Convert column names to lowercase
+  select(ends_with("cd"))   # Keep only columns ending with "cd"
+
+dep_child_age_data <- dep_child_age_data %>%
+  left_join(lookup_data, by = "oa21cd")
+
+# Select Leeds' LAD
+dep_child_age_data <- dep_child_age_data %>%
+  filter(lad22cd == "E08000035")
+
+# Define the geographic levels to loop over
+geo_levels <- c("oa21cd", "lsoa21cd", "msoa21cd")
+
+# Loop over each geography level and store results in a list
+results <- lapply(geo_levels, function(geo) {
+  if (geo %in% colnames(dep_child_age_data)) {  # Check if the column exists
+    dep_child_age_data %>%
+      group_by(across(all_of(geo))) %>%
+      summarise(
+        total_number_of_households = sum(Observation, na.rm = TRUE),
+        eligible = sum(Observation[dep_child_category %in% c(1, 2)], na.rm = TRUE),
+        share_of_eligible = 100 * eligible / total_number_of_households,
+        .groups = "drop"  # Ensure ungrouping after summarising
+      ) %>%
+      select(all_of(geo), total_number_of_households, eligible, share_of_eligible)  # Keep only required columns
+  } else {
+    message(paste("Skipping:", geo, "not found in dataset"))
+    return(NULL)  # Return NULL if the column does not exist
+  }
+})
```

Comment:
This is for creating new aggregate statistics for each geographic unit (e.g. LSOA), for each level of aggregation in geo_levels?

Reply:
Could we, in addition, also have the more granular info, the ratio of the [0-4] and [5-9]?